### PR TITLE
NCES Private School Data Upload

### PR DESCRIPTION
**Pull request by hannahbergam**
Both the School seeding and the script dry run are working locally. I'm going to merge as dry_run: false, but will test live after merging.

## Links
https://docs.google.com/document/d/1CZbauSgOf81C9JbTdorXrNcO1SqccwCCMkNp0ElWRnw/edit?usp=sharing

- spec: []()
- jira ticket: https://codedotorg.atlassian.net/browse/LP-2355

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: bin/oneoff/nces_data/import_school_stats_by_years_2020_2021_pss [hunk 1]
```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env ruby
+
+require_relative '../../../dashboard/config/environment'
+
+CDO.log = Logger.new(STDOUT)
+
+SURVEY_YEAR = '2019-2020'.freeze
```
Comment (bethanyaconnor) [inline bin/oneoff/nces_data/import_school_stats_by_years_2020_2021_pss]: should this be "2020-2021"?

Comment (bethanyaconnor) [inline bin/oneoff/nces_data/import_school_stats_by_years_2020_2021_pss]: should this be "2020-2021"?

File: bin/oneoff/nces_data/import_school_stats_by_years_2020_2021_pss [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  '42' => 'rural_distant',
+  '43' => 'rural_remote'
+}.freeze
+
+AWS::S3.process_file('cdo-nces', "2020-2021/pss/final_schools_private.csv") do |filename|
+  SchoolStatsByYear.merge_from_csv(filename, {col_sep: ",", headers: true, quote_char: "\x00", encoding: 'UTF-8'}, dry_run: true) do |row|
+    {
+      school_id:          row['School ID - NCES Assigned [Private School] Latest available year'],
+      school_year:        SURVEY_YEAR,
+      grades_offered_lo:  row['Lowest Grade Taught [Private School] 2019-20'].presence.try {|v| GRADES_MAP[v]},
```
Comment (bethanyaconnor) [inline bin/oneoff/nces_data/import_school_stats_by_years_2020_2021_pss]: Should this be 2020-21? (here and below)

Comment (bethanyaconnor) [inline bin/oneoff/nces_data/import_school_stats_by_years_2020_2021_pss]: Should this be 2020-21? (here and below)

Comment (bethanyaconnor) [reply bin/oneoff/nces_data/import_school_stats_by_years_2020_2021_pss]: Nevermind, I see it's 2019-20 in the csv on drive

Comment (bethanyaconnor) [reply bin/oneoff/nces_data/import_school_stats_by_years_2020_2021_pss]: Nevermind, I see it's 2019-20 in the csv on drive

File: bin/oneoff/nces_data/import_school_stats_by_years_2020_2021_pss (hunk 1 above)
Comment (hannahbergam) [reply bin/oneoff/nces_data/import_school_stats_by_years_2020_2021_pss]: No- it's actually the 2019-2020 data. I'm not sure why we didn't add it this year, or if it was only just released. I originally thought this too until I opened up the CSV files and saw that nearly every column was labeled with the years😂

Comment (hannahbergam) [reply bin/oneoff/nces_data/import_school_stats_by_years_2020_2021_pss]: (Answered above)

File: dashboard/app/models/school.rb [hunk 3]
```diff
@@ -382,6 +382,27 @@ def self.seed_from_s3
           }
         end
       end
+
+      CDO.log.info "Seeding 2019-2020 private school data."
+      AWS::S3.seed_from_file('cdo-nces', "2020-2021/pss/schools_private.csv") do |filename|
+        merge_from_csv(filename, {headers: true, encoding: 'ISO-8859-1:UTF-8'}, true, is_dry_run: false) do |row|
```
Comment (bethanyaconnor) [inline dashboard/app/models/school.rb:388]: do you want this `is_dry_run` to be true for now?

File: bin/oneoff/nces_data/import_school_stats_by_years_2020_2021_pss (hunk 1 above)
Comment (bethanyaconnor) [reply bin/oneoff/nces_data/import_school_stats_by_years_2020_2021_pss]: Okay sounds good! (Also I don't know why github was posting my comments multiple times...)

File: dashboard/app/models/school.rb (hunk 3 above)
Comment (hannahbergam) [reply dashboard/app/models/school.rb:388]: I will update this before merging!

File: bin/oneoff/nces_data/import_school_stats_by_years_2020_2021_pss (hunk 1 above)
Comment (hannahbergam) [reply bin/oneoff/nces_data/import_school_stats_by_years_2020_2021_pss]: I just saw your comment that you looked at the csv too:)

**Review by bethanyaconnor — APPROVED** [review]
This looks good from my limited knowledge here!

**Pull request merged**